kategory-docs/docs/docs/effects/async/README.md

```diff
@@ -4,6 +4,68 @@ title: Async
 permalink: /docs/effects/async/
 ---
 
-## Async
+## AsyncContext
 
-TODO
\ No newline at end of file
+Being able to run code in a different context of execution (i.e. thread) than the current one implies that, even if it's part of a sequence, the code will have to be asynchronous.
+Running asynchronous code always requires a callback after completion on error capable of returning to the current thread.
+
+The same way the typeclass [`Monad`]({{ '/docs/typeclasses/monad' | relative_url }}) represents a sequence of events, and [`MonadError`]({{ '/docs/typeclasses/monaderror' | relative_url }}) a sequence that can fail, the typeclass `AsyncContext` represents asynchronous code with a callback.
+Examples of that can run code asynchronously are typically datatypes that can suspend effects, and delay evaluation.
+
+```kotlin:ank
+import kategory.*
+import kategory.effects.*
+
+IO.asyncContext()
+  .runAsync { callback: (Either<Throwable, Int>) -> Unit -> 
+    callback(1.right()) 
+  }
+```
+
+### Main Combinators
+
+#### runAsync
+
+Receives a function returning unit with a callback as a parameter.
+The function is responsible of calling the callback once it obtains a result.
+The callback accepts `Either<Throwable, A>` as the return, where the left side of the [`Either`]({{ '/docs/datatypes/either' | relative_url }}) represents an error in the execution and the right side is the completion value of the operation.
+
+### Syntax
+
+#### (() -> A)#runAsync
+
+Runs the current function in the AsyncContext passed as a parameter.
+
+Note that there is no automatic error handling or wrapping of exceptions.
+
+#### (() -> Either<Throwable, A>)#runAsyncUnsafe
+
+Runs the current function in the AsyncContext passed as a parameter.
+
+While there is no wrapping of exceptions, the left side of the [`Either`]({{ '/docs/datatypes/either' | relative_url }}) represents an error in the execution.
+
+#### binding#bindAsync
+
+While in a binding block of a Monadic Comprehension, bindAsync runs a function parameter in the AsyncContext passed as a parameter,
+and then awaits for the result before continuing the execution.
+
+Note that there is no automatic error handling or wrapping of exceptions.
+
+#### binding#bindAsyncUnsafe
+
+While in a binding block of a Monadic Comprehension, bindAsync runs a function parameter in the AsyncContext passed as a parameter,
+and then awaits for the result before continuing the execution.
+
+While there is no wrapping of exceptions, the left side of the [`Either`]({{ '/docs/datatypes/either' | relative_url }}) represents an error in the execution.
+
+### Laws
+
+Kategory provides [`AsyncLaws`]({{ '/docs/typeclasses/laws#asynclaws' | relative_url }}) in the form of test cases for internal verification of lawful instances and third party apps creating their own `AsyncContext` instances.
+
+### Data types
+
+The following datatypes in Kategory provide instances that adhere to the `AsyncContext` typeclass.
```

Comment:
We should probably come up with the inverse function we are using to show instances for data types but for type classes so that this documentation does not get outdated as the number of instances increase